happi/cli.py

```diff
@@ -873,6 +884,125 @@ def no_create_channel(*args, **kwargs):
+    happi.SearchResult as an positional argument and returns None if
+    successful.
+
+    To import additional checks, provide a file with your check function
+    and a list named ``checks`` containing the desired functions.
+    """
+    logger.debug('Starting audit block')
+    from .audit import checks, verify_result
+
+    # if a file is provided, make its functions available
+    if ext_file:
+        fp = Path(ext_file)
+        sys.path.insert(1, str(fp.parent))
+        import importlib
+        ext_module = importlib.import_module(fp.stem)
+        ext_checks = getattr(ext_module, 'checks')
+        checks.extend(ext_checks)
+
+    # List checks subcommand
+    if list_checks:
+        check_pt = prettytable.PrettyTable(field_names=['name',
+                                                        'description'])
+        check_pt.hrules = prettytable.ALL
+        check_pt.align['description'] = 'l'
+        for chk in checks:
+            check_pt.add_row([chk.__name__, inspect.cleandoc(chk.__doc__)])
+        print(check_pt)
+        return
+
+    # gather selected checks
+    if check_choices:
+        check_list = []
+        for check_name in check_choices:
+            # check if provided check name is a substring of any checks
+            matches = [fn for fn in checks if check_name in fn.__name__]
+            if len(matches) != 1:
+                raise click.BadParameter(
+                    f'provided check name ({check_name}) must match only'
+                    f'one check.  Matches: ({[ch.__name__ for ch in matches]})'
+                )
+            check_list.append(matches[0])
+    else:
+        # take all checks
+        check_list = checks
+
+    client: happi.Client = ctx.obj
+
+    results = search_parser(client, use_glob, search_criteria)
+    click.echo(f'found {len(results)} items to verify')
+    click.echo(f'running checks: {[f.__name__ for f in check_list]}')
+
+    test_results = {'name': [], 'success': [], 'msg': []}
+    f = io.StringIO()
+    for i, res in enumerate(results):
+        print(f'checking device #: {i}', end='\r')
+        # Capture stdout, stderr for this audit
+        with redirect_stderr(f), redirect_stdout(f):
+            success, msg = verify_result(res, check_list)
+            test_results['name'].append(res.item.name)
+            test_results['success'].append(success)
+            test_results['msg'].append(msg)
+
+    # print outs
+    if names_only:
+        click.echo('\n')
+        click.echo(' '.join(test_results['name']))
+    else:
+        pt = prettytable.PrettyTable(field_names=['name', 'check', 'error'])
+        for name, success, msg in zip(test_results['name'],
+                                      test_results['success'],
+                                      test_results['msg']):
+            if not success:
+                check_text, error = msg.split(': ', 1)
+                check_name = re.search(r'\((.*)\)', check_text)[1]
```

Comment:
It seems odd to me that you need to regex through your own return value to get this info, I'd imagine this would be simpler to embed in the test_results

Reply:
erm.  Yes.  That makes more sense..